test/unit/source_tests.rb

```diff
@@ -99,6 +107,32 @@ class RenderTests < InitTests
       assert_equal exp, subject.partial(@template_name, @file_locals)
     end
 
+    should "only render templates with the matching ext if one is specified" do
+      source = @source_class.new(@root, :ext => 'nm')
+      file_path = Factory.template_file('locals.nm')
+      exp = Nm::Template.new(source, file_path, @file_locals).__data__
+      ['locals', 'locals.nm'].each do |name|
+        assert_equal exp, source.render(name, @file_locals)
+      end
+
+      source = @source_class.new(@root, :ext => 'inem')
+      file_path = Factory.template_file('locals_alt.data.inem')
+      exp = Nm::Template.new(source, file_path, @file_locals).__data__
+      ['locals', 'locals_alt', 'locals_alt.data', 'locals_alt.data.inem'].each do |name|
+        assert_equal exp, source.render(name, @file_locals)
+      end
+
+      source = @source_class.new(@root, :ext => 'nm')
+      ['locals_alt', 'locals_alt.data', 'locals_alt.data.inem'].each do |name|
+        assert_raises(ArgumentError){ source.render(name, @file_locals) }
+      end
+
+      source = @source_class.new(@root, :ext => 'data')
+      ['locals_alt', 'locals_alt.data', 'locals_alt.data.inem'].each do |name|
+        assert_raises(ArgumentError){ source.render(name, @file_locals) }
+      end
```

Comment:
@jcredding FYI, I had to update these tests now that I'm raising an exception for non-existing files.